content/posts/2020-10-29-loadsmart-take-home-exercises/index.md

```diff
@@ -0,0 +1,67 @@
+---
+title: Loadsmart Take Home Exercises
+author: Flavio S. Truzzi
+twitter: flaviotruzzi
+layout: post
+lang: en
+path: /blog/loadsmart-take-home-exercises
+date: '2020-10-29'
+comments: true
+---
+
+Hiring is something taken very seriously at Loadsmart. One of our principles states that **Our biggest asset is our People**, we invest a lot of time in our hiring process. Our process for engineering includes what we call a Take-Home Exercise, and today I would like to give some tips for doing well on these.
+
+Our exercises are tailored for different positions, so we have different exercises for Backends, Frontends, SREs, and Data Scientists. Most of the tips can be applied to all of them.
+
+### Before starting the exercise
+
+Carefully read the file we send. There we have the goal and the expectations. A lot of times, we receive exercises that do not achieve the requirements specified in the text. Focus on what we ask, this will change according to the task, but generally, we will look into:
+
+1. Your ability to code;
+2. Written documentation;
+3. Domain-Specific knowledge;
+4. Automated tests.
+
+For Data Science tests, we care a lot to your:
+
+1. Line of Thought - looking into why you took the decisions you took;
+2. Feature Engineering and the ideas behind them;
+3. Project structure and architectural aspects.
+
+Make sure you understand what is being asked, and don’t be afraid to reach out with questions, but try to keep them concise.
+
+### Starting your exercise
+
+Start your project with:
+
+```bash
+mkdir take-home-exercise && cd take-home-exercise
+git init
+touch README.md
+```
+
+Having a git repository helps us to understand how you work, your priorities, and how you designed your solution.
+Make sure your README.md contains instructions on how to run your code, and how to run your automated tests.
+
+### Explicit requirements
+
+Here at Loadsmart, we mainly use 3 different programming languages. JavaScript for frontend, and Python and Golang for the backend. At this moment, we are not asking for Golang exercises, but that may change in the future. If that happens, please make sure you have your go module setup. For python, make sure you write your requirements.txt (or another control file like pipenv or poetry), and for JavaScript package.json.
```

Comment:
As a suggestion, we can add TypeScript because, in new projects and libraries, we are creating with it.
 
```suggestion
Here at Loadsmart, we mainly use 4 different programming languages. TypeScript and JavaScript for frontend, and Python and Golang for the backend. At this moment, we are not asking for Golang exercises, but that may change in the future. If that happens, please make sure you have your go module setup. For python, make sure you write your requirements.txt (or another control file like pipenv or poetry), and for JavaScript package.json with yarn.lock.
```